index.js

```diff
@@ -443,58 +459,93 @@ class HtmlWebpackInlineSVGPlugin {
+        // create a new instance of SVGO
+        // passing it the merged config, to optimize the svg
 
-            // read in the svg
+        const svgo = new SVGO(config)
 
-            fs.readFile(path.resolve(this.outputPath, svgSrc), 'utf8', (err, data) => {
+        svgo.optimize(data)
+            .then((result) => {
+
+                const optimisedSVG = result.data
 
-                if (err) reject(err)
+                html = this.replaceImageWithSVG(html, inlineImage, optimisedSVG)
 
-                const configObj = Object.assign(svgoDefaultConfig, this.userConfig)
+                resolve(html)
 
-                const config = {}
+            })
+            .catch((err) => console.log(chalk.red(err)))
+    }
 
 
-                // pass all objects to the config.plugins array
+    /**
+     * append the inlineImages SVG data to the output HTML and remove the original img by
+     * loading the SVG data from the filesystem or from an URL
+     * @param {string} html
+     * @param {Object} inlineImage - parse5 document
+     * @returns {Promise}
+     *
+     */
+    processOutputHtml (html, inlineImage) {
 
-                config.plugins = _.map(configObj, (value, key) => ({ [key]: value }));
+        return new Promise((resolve, reject) => {
 
+            const svgSrc = this.getImagesSrc(inlineImage)
 
-                // create a new instance of SVGO
-                // passing it the merged config, to optimize the svg
 
-                const svgo = new SVGO(config)
+            // if the image isn't valid resolve
+            if (!svgSrc) return resolve(html)
 
-                svgo.optimize(data)
-                    .then((result) => {
+            // read in the svg
+            fs.readFile(path.resolve(this.outputPath, svgSrc), 'utf8', (err, data) => {
+                if (!err) {
+                    this.optimizeSvg({ html, inlineImage, data, resolve })
+                    return
+                }
 
-                        const optimisedSVG = result.data
+                // loading from the filesystem failed
+                if (!this.allowFromUrl) {
+                    reject(err)
+                    return
+                }
 
-                        html = this.replaceImageWithSVG(html, inlineImage, optimisedSVG)
+                axios.get(svgSrc)
+                    .then(({ status, data }) => {
+                        if (status !== 200) {
+                            new Error(data);
+                        }
 
-                        resolve(html)
+                        this.optimizeSvg({ html, inlineImage, svgSrc, data, resolve })
+                    })
+                    .catch(() =>  {
+                        const start = inlineImage.sourceCodeLocation.startOffset
 
+                        // Add the failed attribute so in next iteration this image gets filtered
+                        const htmlWithFailedImage = html.substring(0, start) + '<img failed ' + html.substring((start + 5))
```

Comment:
This attribute labelling is clever. Given the complexity added by the asynchronous Axios promise it doesn't occur to me a better way of doing it.

Can you use the `error` parameter coming from the `catch` statement to print a warning on console?
```js
catch(error) {
  console.warn('html-webpack-inline-svg-plugin: An image URL failed to download:', error)
  // more code...
}
```

Reply:
I unsure about the logic inside this `catch` statement. I think when a local file fails to load the entire compilation fails with an `Error: ENOENT: no such file or directory`. Hence, same should happen if the HTTP GET call fails for any reason.

You can reproduce an erroneous local reading by changing on `spec\fixtures\index-allow-from-url.html` line 31 to `src="images/faceDFDFDF.svg"`. Then the test will fail becaure there has been an `ENOENT` error.

Reply:
I don't think a build process should fail because of a third party system providing an asset via Network. I'm afraid we don't agree here.

This suggestion, on the other hand, would require me to change the error handling and tests too.

Reply:
It's just to keep consistency with what was in place. It would be a bit estrange that, in case of error, for local files the webpack build failed but it wouldn't for files retrieved from URL.

I do think however that adding some `failSilenty` functionality to the project would be something interesting to consider.

I've added an explanatory text to `README.md` to explain this behaviour that it wasn't very clear actually.